lib/frameworks/mocha.js

```diff
@@ -18,7 +19,7 @@ module.exports.run = function(config, specs, capabilities) {
         requires = config.mochaOpts.require,
         runner;
 
-    if(typeof config.mochaOpts.ui !== 'string' || !config.mochaOpts.ui.match(/(bdd|tdd)/i)) {
+    if(typeof config.mochaOpts.ui !== 'string') {
```

Comment:
Because writing it this way makes it so that the worst that happens for interfaces you don't support is that generators won't work. The worst case for the way you wrote it is that it reverts to an entirely different interface.